src/kudu/rpc/rpc-test.cc

```diff
@@ -1360,5 +1366,42 @@ TEST_P(TestRpc, TestCancellationMultiThreads) {
   client_messenger->Shutdown();
 }
 
+TEST_F(TestRpc, TestCallWithNormalTLS) {
+  FLAGS_enable_normal_tls = true;
```

Comment:
This tests that TLS Client <-> TLS server works as expected, but we also need a test for a non-TLS client to a TLS server to exercise the migration path.

This means that FLAGS_enable_normal_tls needs to be set for the server, and it needs to be unset for the client. Since this gtest runs in a single process, this is impossible to do.

You should add `ClientNegotiation::setNormalTLS(bool flag)` and `ServerNegotiation::setNormalTLS(bool flag)` and write `ClientNegotation::Negotiate` and `ServerNegotiation::Negotiate` in terms of a member variable flag.

This will allow you to test `ClientNegotiation::setNormalTLS(false)` and `ServerNegotiation::setNormalTLS(true)`.

`FLAGS_enable_normal_tls` can be used to set the default value of that flag.